src/BaseFirestoreRepository.ts

```diff
@@ -125,6 +127,9 @@ export default class BaseFirestoreRepository<T extends IEntity>
     if (limitVal) {
       query = query.limit(limitVal);
     }
-    return query.get().then(this.extractTFromColSnap);
+
+    const queryRes = await query.get();
+
+    return this.extractTFromColSnap(queryRes);
```

Comment:
I don't see why removing the other implementation is better than declaring a variable that will only be used once.

Reply:
consistency (`async/await` vs `.then/.catch`) and ease of debugging intermediate return types.

Reply:
I understand the consistency part but when it makes sense and about the `debugging intermediate` the first param of `extractTFromColSnap` was the response of query.get.